### DISCUSSION TITLE
Using HostedGPT for business?

### DISCUSSION THREAD

**Discussion by keithschacht** [AllYourBot/hostedgpt#336] [General]
I just had a great conversation with this guy Jon who has been using a different chat bot front-end. He runs a consulting company that helps business figure out how they can use AI / chatbots for their business and then he sets them up with an implementation that he'll customize. The current project he's been using is moving a bit sideways so he's looking for a more solid foundation to build upon. We compared notes and he was excited by how this HostedGPT project is evolving so he's going to see if he can build his business on this.

I'm curious to ask: @stephan-buckmaster, @stevenlafl, @florianfelsing, @olimart, @jasonpaulso, @al3rez, @anandvc  — are any of you using HostedGPT for your own business or do yo know of anyone using this project for internal business use? I told Jon to figure out all the features he needs so that he can move his clients over to this platform and I can take that into consideration as I'm prioritizing efforts. But I just wanted to share this as a general FYI in case there's anyone else who has a particular interest in using HostedGPT with employees / internal teams.

**Comment by stevenlafl** [comment]
That's right, it is a potential use case. While I am not ready to propose it to my company just yet, that is one reason I submitted the Kubernetes template RFC. With some modifications it could go into a secure production environment.

I would suggest to Jon that he might be interested in seeking SOC 2 certification, which would allow him to become a SaaS vendor for  organizations with stricter standards such as with local government and NGOs. The code and IaC could go through a feedback loop with the auditor and this project's maintainers - otherwise, he'll have to fork it and maintain a new set of edits.

For his case, it might make sense to add one more layer: sub-organizations (or business-units). Folks like analytics, and knowing which segment of your business uses AI the most could be useful insight.

**Comment by stephan-buckmaster** [comment]
We're using it internally, as a business, just starting out. There could be
interest from clients, working on finding out what they need.

The app does have a nice design! Feature I'm hoping to add (soon) is to
connect to our own LLM server.

Stephan


On Wed, 8 May 2024 at 11:25, Steven Linn ***@***.***> wrote:

> That's right, it is a potential use case. While I am not ready to propose
> it to my company just yet, that is one reason I submitted the Kubernetes
> template RFC. With some modifications it could go into a secure production
> environment.
>
> I would suggest to Jon that he might be interested in seeking SOC 2
> certification, which would allow him to become a vendor for organizations
> with stricter standards such as with local government and NGOs. The code
> and IaC could go through a feedback loop with the auditor and this
> project's maintainers - otherwise, he'll have to fork it and maintain a new
> set of edits.
>
> For his case, it might make sense to add one more layer: sub-organizations
> (or business-units). Folks like analytics, and knowing which segment of
> your business uses AI the most could be useful insight.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/allyourbot/hostedgpt/discussions/336#discussioncomment-9358833>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AACAV3RSMJDP32IATOBPZILZBJUYBAVCNFSM6AAAAABHNN2F4SVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TGNJYHAZTG>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Stephan Wehner, Ph.D.
The Buckmaster Institute, Inc.
2150 Adanac Street
Vancouver BC V5L 2E7
Canada
Cell (604) 767-7415
Fax (888) 808-4655

Sign up for our free email course
http://buckmaster.ca/small_business_website_mistakes.html

http://www.buckmaster.ca
http://answer4img.com
http://loggingit.com
http://clocklist.com
http://stephansmap.org
http://benchology.com
http://www.trafficlife.com
http://stephan.sugarmotor.org (Personal Blog)
@stephanwehner (Personal Account)
VA7WSK (Personal call sign)

**Reply by keithschacht** [reply]
How does your own LLM server run? A lot of the ones that run on desktop machines just expose an OpenAI-equivalent API. Is that the case with yours?

One thing Jon mentioned to me verbally was integration with Groq because he uses a few of the models on there. I told him that wouldn't be too hard to do (and maybe I'd prioritize it next, not sure yet).

It'd be nice if we could address Jon's needs and yours at the same time.

**Comment by stephan-buckmaster** [comment]
Yes, I'm looking at a very standard interface. So the ruby-openai gem, that
hostedgpt uses, should work. Just need to be able to configure host, port,
authentication etc. I should have it put together end of next week.

Presumably groq offers the same API. But it wasn't obvious immediately,
when I looked over their website.


On Wed, 8 May 2024 at 11:56, Keith Schacht ***@***.***> wrote:

> How does your own LLM server run? A lot of the ones that run on desktop
> machines just expose an OpenAI-equivalent API. Is that the case with yours?
>
> One thing Jon mentioned to me verbally was integration with Groq because
> he uses a few of the models on there. I told him that wouldn't be too hard
> to do (and maybe I'd prioritize it next, not sure yet).
>
> It'd be nice if we could address Jon's needs and yours at the same time.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/allyourbot/hostedgpt/discussions/336#discussioncomment-9359128>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AACAV3UK4PD4KLCUGXA6F7DZBJYNPAVCNFSM6AAAAABHNN2F4SVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TGNJZGEZDQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Stephan Wehner, Ph.D.
The Buckmaster Institute, Inc.
2150 Adanac Street
Vancouver BC V5L 2E7
Canada
Cell (604) 767-7415
Fax (888) 808-4655

Sign up for our free email course
http://buckmaster.ca/small_business_website_mistakes.html

http://www.buckmaster.ca
http://answer4img.com
http://loggingit.com
http://clocklist.com
http://stephansmap.org
http://benchology.com
http://www.trafficlife.com
http://stephan.sugarmotor.org (Personal Blog)
@stephanwehner (Personal Account)
VA7WSK (Personal call sign)

**Reply by keithschacht** [reply]
Ah, got it. That’s a good idea. Are you planning to do it as part of your current PR or a new one? I suspect you mean a new one so I tee’d this up. Let me know if this captures what you’re thinking and comment on it so I can assign to you:
https://github.com/allyourbot/hostedgpt/issues/341

Or if you’re doing as part of your current PR then I can just remove this new issue.

**Comment by stephan-buckmaster** [comment]
Separate, in fact even the present one would have some natural follow-ups,
I think ... One thing at a time, we'll see
